app/controllers/user_profiles_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+class UserProfilesController < ApplicationController
+  before_action :set_user, only: [:show, :update, :destroy]
```

Comment:
This line could be written as `before_action :set_user` as every action in the controller needs a `user` object to work with. Even in the long run, if you add any new actions to this controller, those will always require a `user` object. So, IMO, it is totally fine having

```ruby
before_action :set_user
```

Less keystrokes :wink: